jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -620,6 +640,22 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
     return sign ? -ECMA_NUMBER_ZERO : ECMA_NUMBER_ZERO;
   }
 
+  /** If the convertable number is bigger than Number.MAX_VALUE
+    * then return with +- Infinity */
+  if (!e_in_lit_sign && (e > NUMBER_MAX_VALUE_EXPONENT ||
+      (e == NUMBER_MAX_VALUE_EXPONENT && fraction_uint64 > NUMBER_MAX_VALUE_BASE)))
```

Comment:
This condition (the second part of the ||) and the similar one below don't seem right. `fraction_uint64` is an uint64_t while `NUMBER_MAX_VALUE_BASE` is a floating point number (less than 2). I think that the value of `NUMBER_MAX_VALUE_BASE` (and `NUMBER_MIN_VALUE_BASE` below) should be updated to the appropriate uint64 literals.

Moreover, this again shows that the test is still not enough.We should have a test case with `e` equaling `NUMBER_MAX_VALUE_EXPONENT` and `fraction_uint64` above and below the `NUMBER_MAX_VALUE_BASE` threshold. (BTW, is the name `NUMBER_MAX_VALUE_BASE` still appropriate?))

Reply:
Yesterday evening I believed you that this relation was not correct, and should be reversed but I've just realized (after a half an hour of thinking) that it was correct. Playing around zero is really tricky. We know that `5e-324` is the `Number.MIN_VALUE`. So lets examine this number's "neighbours". These are e.g. `4e-324` and `6e-324`. The `4e-324` is smaller than `6e-324` but `6e-324` is closer to zero than `4e-324`. So it means if the `fraction_uint64` is `4` and `e` is `-324` the result should be this number, but if the `fraction_uint64` is `6` and `e` is `-324` it should be zero. So IMHO the relation is correct.

BTW I realized it also needs a bit of a change because  `fraction_uint64` should be normalized before this if statement, so I'll update this PR soon.

Reply:
> These are e.g. 4e-324 and 6e-324. The 4e-324 is smaller than 6e-324 but 6e-324 is closer to zero than 4e-324.

Could you please elaborate on this?... I know no version of math where 0 < 6N < 4N for any positive N. And 1/10^324 is a positive number. If you take 6 units of a very small something, it will still be more than 4 units of the very same very small something. Or I get something completely wrong in your reasoning.

> fraction_uint64 should be normalized

I'm not sure yet what you mean by normalization but do not convert it to floating point. I do think that the MIN / MAX values have to be updated to be integer values somehow.

Reply:
Now I'm seriously confused. But your reasoning seems correct.  I'm going to rethink it again.